codegen/src/lib.rs

```diff
@@ -239,13 +238,11 @@ use self::macro_path::MacroPath;
 ///     }
 /// }
 ///
-/// #[delegate(derive(for<> AsInner<str>))]
+/// #[delegate(derive(AsInner<str>))]
 /// struct FirstName(String);
 ///
 /// #[delegate(derive(
-///     for<I> AsInner<str>
-///     where
-///         I: AsInner<str> + 'static;
+///     AsInner<str> where I: AsInner<str> + 'static;
```

Comment:
@50U10FCA7 here and in README, we better rework this example to show the case when the `for<..>` syntax really is needed. 

Slightly extending the added `trait_and_type_generic_parameter` test, I think, should be sufficient.